### PR TITLE
Feature/nullable reference types

### PR DESCRIPTION
**Pull request by jared-improbable**

### CODE REVIEW

**Review by zeroZshadow — APPROVED** [review]

File: Workers/DatabaseSyncWorker/Reflection.cs [hunk 1]
```diff
@@ -55,16 +55,33 @@ public HydrationType(Hydration.HydrateDelegate hydrate, Func<object, string> pro
...
                     var fromSchemaData = method.DeclaringType.GetMethods(BindingFlags.Public | BindingFlags.Static).First(IsGetProfileIdFromSchemaData);
 
                     var hydrateDelegate = (Hydration.HydrateDelegate) method.CreateDelegate(typeof(Hydration.HydrateDelegate));
 
 
-                    var type = new HydrationType(hydrateDelegate, instance => (string) property.GetMethod.Invoke(instance, null), fields =>
+                    var type = new HydrationType(hydrateDelegate, instance =>
+                    {
+                        var result = property.GetMethod.Invoke(instance, null);
+                        return result == null ? "" : (string) result;
```
Comment (jamiebrynes7) [inline Workers/DatabaseSyncWorker/Reflection.cs:79]: Might be able to do:

```
return result as string ?? "";
```

depending on the behaviour of the `as` cast

File: Workers/DatabaseSyncWorker/DatabaseSyncLogic.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public class DatabaseSyncLogic : IDisposable
         private readonly WhenAllComponents whenWorker = new WhenAllComponents(Worker.ComponentId);
         private readonly WhenAllComponents whenWorkerClient = new WhenAllComponents(Worker.ComponentId, PlayerClient.ComponentId);
 
-        private static IReadOnlyDictionary<uint, Reflection.HydrationType> restoreComponents;
+        private static IReadOnlyDictionary<uint, Reflection.HydrationType>? _restoreComponents;
```
Comment (jamiebrynes7) [inline Workers/DatabaseSyncWorker/DatabaseSyncLogic.cs]: Why the `_` prefix?

**Review by jamiebrynes7 — APPROVED** [review]

File: Workers/DatabaseSyncWorker/Reflection.cs (hunk 1 above)
Comment (jared-improbable) [reply Workers/DatabaseSyncWorker/Reflection.cs:79]: Nice, did that

File: Workers/DatabaseSyncWorker/DatabaseSyncLogic.cs (hunk 2 above)
Comment (jared-improbable) [reply Workers/DatabaseSyncWorker/DatabaseSyncLogic.cs]: Mm, default resharper rules

**Pull request merged**